### ISSUE TITLE
Latest beta - interaction with Ground Construction...?

### ISSUE THREAD

**Issue by AccidentalDisassembly** [net-lisias-ksp/TweakScale#141]
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/5215977/KSP.log)
Error preventing the game from loading, moderately modded install - 

PartLoader: Part 'Squad/Parts/Command/Mk1-3Pod/mk1-3/mk1-3pod' has no database record. Creating.
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

[DragCubeSystem]: Drag cubes not found or cannot be read for part Part. Generating New drag cubes.
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

DragCubeSystem: Creating drag cubes for part 'mk1-3pod'
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug/Debug.bindings.h Line: 35)

Uploading Crash Report
NullReferenceException: Object reference not set to an instance of an object
  at TweakScale.TweakScale.IPartCostModifier.GetModuleCost (System.Single defaultCost, ModifierStagingSituation situation) [0x0000e] in <452e43633b764e0092a748620c30084b>:0 
  at Part.GetModuleCosts (System.Single defaultCost, ModifierStagingSituation sit) [0x00046] in <c1858a3f77504bd1aaa946fdccf84670>:0 
  at AT_Utils.PartExtensions.TotalCost (Part p) [0x0000e] in <cc0dfa428eee4824909c2512b8440243>:0 
  at AT_Utils.Metric.partsBounds (System.Collections.Generic.IList`1[T] parts, UnityEngine.Transform refT, System.Boolean compute_hull, System.Boolean exclude_disabled) [0x0033b] in <cc0dfa428eee4824909c2512b8440243>:0 
  at AT_Utils.Metric..ctor (Part part, System.Boolean compute_hull, System.Boolean world_space) [0x00032] in <cc0dfa428eee4824909c2512b8440243>:0 
  at AT_Utils.Metric.Volume (Part part) [0x00000] in <cc0dfa428eee4824909c2512b8440243>:0 
  at GroundConstruction.GroundWorkshop.compute_part_efficiency () [0x0001f] in <153e62ae099745349276ac22b21d61e0>:0 
  at GroundConstruction.GroundWorkshop.GetInfo () [0x00008] in <153e62ae099745349276ac22b21d61e0>:0 
  at PartLoader.CompilePartInfo (AvailablePart newPartInfo, Part part) [0x0024c] in <c1858a3f77504bd1aaa946fdccf84670>:0 
  at PartLoader+<CompileParts>d__56.MoveNext () [0x006ea] in <c1858a3f77504bd1aaa946fdccf84670>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00027] in <5aeafee3fea24f37abd1315553f2cfa6>:0 
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
 
(Filename: <452e43633b764e0092a748620c30084b> Line: 0)

**Comment by 7ranceaddic7** [comment]
I don't think this isn't going to be a TS issue.
However, to make that we'll need a KSP.log to go through, first.

**Comment by Lisias** [comment]
There's a flaw on Beta .20 on handling some GameEvents, and this flaw just happened exactly before the Exception @AccidentalDisassembly  posted.

I'm failing to understand how that flaw could be the reason for this second one, but since I don't use AT-Utils on the development installment, I can't rule this out neither.

@AccidentalDisassembly , I'm fixing a stupid mistake on handling costs from parts without resources, and a release .21 is imminent.

Rollback to TS Beta .19 for some days.

I will keep this open until .21 is on the wild and I have evidences this was fixed.

**Comment by Lisias** [comment]
@AccidentalDisassembly , try the latest beta so see how your issue behaves.

https://github.com/net-lisias-ksp/TweakScale/releases/tag/PRERELEASE%2F2.5.0.23

Thanks!

**Comment by Lisias** [comment]
@AccidentalDisassembly , please try the new latest beta so see how your issue behaves.

https://github.com/net-lisias-ksp/TweakScale/releases/tag/PRERELEASE%2F2.5.0.24

Thanks!

**Comment by AccidentalDisassembly** [comment]
> 
> 
> @AccidentalDisassembly , please try the new latest beta so see how your issue behaves.
> 
> https://github.com/net-lisias-ksp/TweakScale/releases/tag/PRERELEASE%2F2.5.0.24
> 
> Thanks!

That specific issue seems to have been resolved, but loading still hangs with an error - this time on the Mk 1-3 pod (it seems). Somewhat modded install; I've been trying to add mods back in to 1.10.1 and I would presume that any of them could be causing this too. It just happens to show an error with TweakScale...
[Player.log](https://github.com/net-lisias-ksp/TweakScale/files/5285937/Player.log)

**Comment by Lisias** [comment]
I just loaded KSP 1.11 using the current [Ground Construction](https://github.com/allista/GroundConstruction/releases/tag/v2.6.4.1) and the current code for the imminent TweakScale 2.4.4.0 and I found no exceptions on the KSP.log neither Player.log and the game loaded fine.

I'm presuming this is something on some other add'on, as the current Ground Construction is older than this issue.

In a way or another, I'm closing this.
